xbmc/cores/RetroPlayer/playback/ReversiblePlayback.cpp

```diff
@@ -224,8 +224,9 @@ bool CReversiblePlayback::LoadSavestate(const std::string& path)
     if (m_gameClient->Deserialize(savestate->GetMemoryData(), memorySize))
     {
       m_totalFrameCount = savestate->TimestampFrames();
-      m_autosavePath = path;
       bSuccess = true;
+      if (savestate->Type() == SAVE_TYPE::AUTO)
+        m_autosavePath = path;
```

Comment:
If I am not wrong there is no other way to save the progress, so if you have loaded a manual save state the progress will never be saved. The only way to save will be to create a new save state

Reply:
I think that the _CreateSavestate(true)_ is called no matter if _m_autosavePath_ is set, so it should work the same way as when you start completely new game and it should create a new autosave in a new slot.

It comes here:
https://github.com/xbmc/xbmc/blob/26e71673dc5937cf31d5ba750a591d51dedee6dc/xbmc/cores/RetroPlayer/playback/ReversiblePlayback.cpp#L184

So the _savePath_ is set empty from the _m_autosavePath_.
And then in the _AddSavestate()_ the empty _savePath_ is handled and it creates a new savestates.
Then the _m_autosavePath_ is set back from the _savePath_ so subsequent saves are saved to the same slot.

Reply:
Also I believe that I've tested this specific behavior, but I'll double check this again tomorrow to be sure.

The bad thing about this is that for every load from a manual savestate you'll get new autosave slot created, so when you load a lot you'll need to delete the bunch of autosaves. I believe it is still better than having your manual savestate rewritten by autosave.

Reply:
So, I've tested it again and it does what was expected (all tests with Autosave feature enabled):
1. When loading from autosave: It rewrites the original autosave every 10 seconds or on exit.
2. If you load from a manual savestate: It will create a new autosave slot after 10 seconds or upon exit.

My only complaint here from the user's perspective is that I don't know whether I am loading a manual savestate or an autosave. How about at least adding caption "Autosave" to autosaves for now?